File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        current_lowest = prices[0]
        max_profits = 0
        for price in prices:
            if price < current_lowest:
                current_lowest = price
            else:
                if price - current_lowest > max_profits:
                    max_profits = price - current_lowest
        return max_profits

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int currentLowest = Integer.MAX_VALUE;
        int maxProfit = 0;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < currentLowest) {
                currentLowest = prices[i];
            } else if (prices[i] - currentLowest > maxProfit) {
                maxProfit = prices[i] - currentLowest;
            }
        }
        return maxProfit;
    }
}
